pkg/services/dashboards/models.go

```diff
@@ -48,6 +49,38 @@ type Dashboard struct {
 	Data  *simplejson.Json
 }
 
+type TimeRange struct {
+	From string
+	To   string
+}
```

Comment:
Can it be more specific type like time.Duration? Or if it can be absolute, we may use interface TimeRange from expr package.

Reply:
The value here is text like `now` and `now-5m`. We'd need to parse it ourselves, but then it's not super convenient for screenshots because we'd need to turn it back into `now` and `now-5m`.

Reply:
`From` and `To` here is text like `now` and `now-5m`. We'd need to parse it ourselves, but then it's not super convenient for screenshots because we'd need to turn it back into `now` and `now-5m`.

Reply:
>we'd need to turn it back into now and now-5m.

I don't think we'd need to turn it back, dashboards accept unix timestamps just fine. That being said, the benefits are not huge either:

- One less TimeRange struct among the many that already exist.
- If there's a delay in rendering, converting it to a timestamp here means the screenshot will be closer to the actual range that caused the alert to fire.

Might not be worth the effort, I'll leave it up to someone with a stronger opinion. That being said, if you want to do it you can follow the example from [publicdashboards.buildTimeSettings](https://github.com/grafana/grafana/blob/bffd55efd400e99c9856d26238c2904af37574e3/pkg/services/publicdashboards/service/query.go#L349-L369)

Reply:
On second thought, why are we setting `To` and `From` at all if we're not going to convert to a timestamp for the purposes of the delay mentioned above?

If we leave them unset, the dashboard view will take care of setting the right defaults and should even correctly handle panel-specific overrides like `timeFrom` and `timeShift`. This seems to be how it's done when constructing the share url in the front-end: https://github.com/grafana/grafana/blob/53186c14a4ccac9b360656d3f46a1892c2b45c05/public/app/features/dashboard/components/ShareModal/utils.ts#L29-L32

Reply:
Should I make `From` and `To` options in `screenshot` package and then we can think about how we want to override this - because no one uses this override feature right now.

Reply:
https://github.com/grafana/grafana/pull/73325